ddtrace/internal/processor/trace.py

```diff
@@ -130,6 +133,21 @@ def process_trace(self, trace):
         return trace
 
 
+@attr.s
+class Trace128bitTraceIdProcessor(TraceProcessor):
+    """Processor that applies trace-level tags to the trace."""
+
+    def process_trace(self, trace):
+        # type: (List[Span]) -> List[Span]
+        if not trace or trace[0].trace_id <= _MAX_UINT_64BITS:
+            return trace
+
+        trace_id_hob = get_hob_from_128bit_traceid_as_hex(trace[0].trace_id)
+        for span in trace:
+            span._meta[_HIGHER_ORDER_TRACE_ID_BITS] = trace_id_hob
```

Comment:
oof.... can we have the agent or intake expand this tag to all spans instead of us doing it?

Seems weird we need to copy everywhere and not just sending once per-payload per-trace.

Reply:
We could make this a "trace level" tag (ie add this tag to span.context.meta), but I wasn't sure if there were edge cases where `Span.context is None` or if the trace id will be reconstructed by the backend/agent before the trace level tags are set. 

I'll ask for clarification on this.

Reply:
having it accessible from the Span is probably fine, I am just wondering if we can reduce the duplicating of the data on every single span, especially since it is a tag.

there should be a way to have this expansion applied downstream to save us the computational overhead/bytes on the wire.